provisioning/baremetal_pod.go

```diff
@@ -258,28 +260,64 @@ func newMetal3InitContainers(info *ProvisioningInfo) []corev1.Container {
 	}
 
 	initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, true))
+	initContainers = append(initContainers, createInitContainerCoreOSIPA(info))
 
 	return injectProxyAndCA(initContainers, info.Proxy)
 }
 
-func createInitContainerIpaDownloader(images *Images) corev1.Container {
+// getRegistryPullSecret fetches the registry config auth file which is then passed
+// as an env variable to the Ironi-Image's configure-coreos-ipa endpoint. Note that
+// we cannot use LocalObjectReference because the secret is in a different NS.
+func getRegistryPullSecret(info *ProvisioningInfo, targetNamespace string) string {
+	// Check whether the Client is present before using it
+	if info.Client != nil {
+		secretClient := info.Client.CoreV1().Secrets(targetNamespace)
+		secret, err := secretClient.Get(context.TODO(), pullSecretName, metav1.GetOptions{})
+		if err != nil {
+			return ""
+		}
+		// Return the value of .dockerconfigjson key
+		return base64.StdEncoding.EncodeToString(secret.Data[openshiftConfigSecretKey])
```

Comment:
There is a reason this data is in a Secret. We are reading it and passing its contents to Ironic. Is that OK?

Reply:
If we don't pass this then the IPA won't be able to pull the ironic-agent image from the "official" registry. It'll fall back to Dmitry's quay.io version, which is not OK for an official release.

Reply:
openshift/ironic-image#196 should take care of this above concern.

Reply:
We will still be able to read it from oc describe pod metal3..., right?

Reply:
I was looking at Secret selectors only work in the current namespace. I didn't find a way to input the value into the env without this method. One potential solution would be to copy the registry secret into the current ns and then use the LocalObjectReference to read it.

Reply:
> We will still be able to read it from oc describe pod metal3..., right?
When I run "oc describe pod <metal3-pod> -n openshift-machine-api", I see this for a secret passed in via an env var:
HTTP_BASIC_HTPASSWD:     <set to the key 'htpasswd' in secret 'metal3-ironic-inspector-password'>  Optional: false

@kirankt maybe this password also shows up in a similar way. Could you pls confirm?

Reply:
No. This API is not available for Secrets outside the current namespace.

Reply:
> I was looking at Secret selectors only work in the current namespace. I didn't find a way to input the value into the env without this method. One potential solution would be to copy the registry secret into the current ns and then use the LocalObjectReference to read it.

Yes, that's the obvious way.

We also should watch that pull secret in case it changes. That is going to be fun...